internal/daemon/controller/handlers/targets/target_service.go

```diff
@@ -319,22 +364,48 @@ func (s Service) ListTargets(ctx context.Context, req *pbs.ListTargetsRequest) (
 		EstItemCount: uint32(listResp.EstimatedItemCount),
 		RemovedIds:   listResp.DeletedIds,
 		ResponseType: respType,
-		SortBy:       "updated_time",
-		SortDir:      "asc",
-	}
-
-	if listResp.RefreshToken != nil {
-		if listResp.RefreshToken.ResourceType != resource.Target {
-			return nil, errors.New(ctx, errors.Internal, op, "refresh token resource type does not match service resource type")
-		}
-		resp.RefreshToken, err = handlers.MarshalRefreshToken(ctx, &pbs.ListRefreshToken{
-			CreatedTime:         timestamppb.New(listResp.RefreshToken.CreatedTime),
-			UpdatedTime:         timestamppb.New(listResp.RefreshToken.UpdatedTime),
-			ResourceType:        pbs.ResourceType_RESOURCE_TYPE_TARGET,
-			GrantsHash:          listResp.RefreshToken.GrantsHash,
-			LastItemId:          listResp.RefreshToken.LastItemId,
-			LastItemUpdatedTime: timestamppb.New(listResp.RefreshToken.LastItemUpdatedTime),
-		})
+		SortBy:       sortBy,
+		SortDir:      "desc",
+	}
+
+	if listResp.ListToken != nil {
+		if listResp.ListToken.ResourceType != resource.Target {
+			return nil, errors.New(ctx, errors.Internal, op, "list token resource type does not match service resource type")
+		}
+		lt := &pbs.ListToken{
+			CreateTime:   timestamppb.New(listResp.ListToken.CreateTime),
+			ResourceType: pbs.ResourceType_RESOURCE_TYPE_TARGET,
+			GrantsHash:   listResp.ListToken.GrantsHash,
+		}
+		switch st := listResp.ListToken.Subtype.(type) {
+		case *listtoken.PaginationToken:
+			lt.Token = &pbs.ListToken_PaginationToken{
+				PaginationToken: &pbs.PaginationToken{
+					LastItemId:         st.LastItemId,
+					LastItemCreateTime: timestamppb.New(st.LastItemCreateTime),
+				},
+			}
+		case *listtoken.StartRefreshToken:
+			lt.Token = &pbs.ListToken_StartRefreshToken{
+				StartRefreshToken: &pbs.StartRefreshToken{
+					PhaseEndTime: timestamppb.New(st.PhaseEndTime),
+					LastUseTime:  timestamppb.New(st.LastUseTime),
+				},
+			}
+		case *listtoken.RefreshToken:
+			lt.Token = &pbs.ListToken_RefreshToken{
+				RefreshToken: &pbs.RefreshToken{
+					PhaseStartTime:     timestamppb.New(st.PhaseStartTime),
+					PhaseEndTime:       timestamppb.New(st.PhaseEndTime),
+					LastUseTime:        timestamppb.New(st.LastUseTime),
+					LastItemId:         st.LastItemId,
+					LastItemUpdateTime: timestamppb.New(st.LastItemUpdateTime),
+				},
+			}
+		default:
+			return nil, errors.New(ctx, errors.Internal, op, "unexpected list token subtype")
+		}
+		resp.ListToken, err = handlers.MarshalListToken(ctx, lt)
```

Comment:
This logic could likely all be in a helper, but we'll know for sure when working on sessions next